src/Components/AddItem.jsx

```diff
@@ -1,5 +1,85 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
+import { useForm } from 'react-hook-form';
+import firebase from 'firebase';
+import { db } from '../lib/firebase';
 
 export default function AddItem() {
-  return <div>Hi from Add an Item View!</div>;
+  const [token, setToken] = useState('test-token');
+  const { register, handleSubmit, reset, errors } = useForm();
+  const onSubmit = (data) => {
+    const currentList = db.collection('lists').doc(token);
+    currentList
+      .update({
+        items: firebase.firestore.FieldValue.arrayUnion({
+          name: data.itemName,
+          frequency: parseInt(data.itemFrequency),
+          lastPurchased: null,
+        }),
+      })
+      .then(() => {
+        alert('Item has been submitted!');
+        reset();
+      });
+  };
+
+  return (
+    <form onSubmit={handleSubmit(onSubmit)}>
+      <br />
+      <label htmlFor="itemName">Item Name: </label>
+      <input
+        name="itemName"
+        defaultValue=""
+        aria-invalid={errors.itemName ? 'true' : 'false'}
+        ref={register({ required: true, minLength: 3 })}
+      />
+      <br />
+      <span style={{ color: 'red' }}>
+        {errors.itemName && errors.itemName.type === 'required' && (
+          <span role="alert">Item name is required.</span>
```

Comment:
This needs emphasis!
```suggestion
          <span role="alert">Item name is required!</span>
```